freezegun/api.py

```diff
@@ -35,6 +39,66 @@
     import copyreg
 
 
+# keep a cache of module attributes otherwise freezegun will need to analyze too many modules all the time
+# start with `None` as the sentinel value.
+# if `{}` (empty dict) was the sentinel value, there's a chance that `setup_modules_cache()` will be called many times
+_GLOBAL_MODULES_CACHE = None
+
+
+def _get_global_modules_cache():
+    global _GLOBAL_MODULES_CACHE
+    # the first call to this function sets up the global module cache. it's expected to be slower than consecutive calls
+    if _GLOBAL_MODULES_CACHE is None:
+        _GLOBAL_MODULES_CACHE = {}
+        _setup_modules_cache()
+    return _GLOBAL_MODULES_CACHE
+
+
+def _get_module_attributes(module):
+    result = []
+    for attribute_name in dir(module):
+        try:
+            attribute_value = getattr(module, attribute_name)
+        except (ImportError, AttributeError, TypeError):
+            # For certain libraries, this can result in ImportError(_winreg) or AttributeError (celery)
+            continue
+        else:
+            result.append((attribute_name, attribute_value))
+    return result
+
+
+def _setup_modules_cache():
+    for mod_name, module in list(sys.modules.items()):
+        # ignore modules from freezegun
+        if mod_name == __name__ or not mod_name or not module:
+            continue
+        _setup_module_cache(module)
+
+
+def _setup_module_cache(module):
+    global _GLOBAL_MODULES_CACHE
+    date_attrs = []
+    all_module_attributes = _get_module_attributes(module)
+    for attribute_name, attribute_value in all_module_attributes:
+        if id(attribute_value) in _real_time_object_ids:
+            date_attrs.append((attribute_name, attribute_value))
+    _GLOBAL_MODULES_CACHE[module.__name__] = (_get_module_attributes_hash(module), date_attrs)
```

Comment:
I encountered this bug:
```
local.virtualenv/lib/python2.7/site-packages/freezegun/api.py:557: in wrapper
    with self:
local.virtualenv/lib/python2.7/site-packages/freezegun/api.py:434: in __enter__
    return self.start()
local.virtualenv/lib/python2.7/site-packages/freezegun/api.py:490: in start
    module_attrs = _get_cached_module_attributes(mod_name, module)
local.virtualenv/lib/python2.7/site-packages/freezegun/api.py:96: in _get_cached_module_attributes
    global_modules_cache = _get_global_modules_cache()
local.virtualenv/lib/python2.7/site-packages/freezegun/api.py:53: in _get_global_modules_cache
    _setup_modules_cache()
local.virtualenv/lib/python2.7/site-packages/freezegun/api.py:75: in _setup_modules_cache
    _setup_module_cache(module)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

module = <AliasModule 'py.error' for 'py._error.error'>

    def _setup_module_cache(module):
        global _GLOBAL_MODULES_CACHE
        #if not hasattr(module, "__name__"):
        #    return
    
        date_attrs = []
        all_module_attributes = _get_module_attributes(module)
        for attribute_name, attribute_value in all_module_attributes:
            if id(attribute_value) in _real_time_object_ids:
                date_attrs.append((attribute_name, attribute_value))
>       _GLOBAL_MODULES_CACHE[module.__name__] = (_get_module_attributes_hash(module), date_attrs)
E       AttributeError: __name__

all_module_attributes = []
date_attrs = []
module     = <AliasModule 'py.error' for 'py._error.error'>

local.virtualenv/lib/python2.7/site-packages/freezegun/api.py:88: AttributeError
```
It seems ok if I add the following code at the beginning of the function:
```python
def _setup_module_cache(module):
    if not hasattr(module, "__name__"):
        return
    ...
```